config/routes.rb

```diff
@@ -1,12 +1,19 @@
 
 Rails.application.routes.draw do
-  # The priority is based upon order of creation: first created -> highest priority.
-  # See how all your routes lay out with "rake routes".
+  resources :cards
+  resources :users, only: [:new, :create]
+  resources :sessions, only: [:new, :create, :destroy]
 
-  # You can have the root of your site routed with "root"
-  root 'cards#index'
-  get '/all' => 'cards#all'
+  get '/sign_up', to: 'users#new', as: :sign_up
+  get '/log_in', to: 'sessions#new', as: :log_in
+  delete '/log_out', to: 'sessions#destroy', as: :log_out
+
+  get '/secret', to: 'cards#all', as: :secret
+  get '/index' => 'cards#index'
   post '/check' => 'cards#check'
+  root to: 'cards#index'
+  get 'profile/:id/edit' => 'profile#edit', as: :edit_profile
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.